Covenant Training Pistol

Boltersam opened the pull request:
Adds a recoloured plasma pistol with TTR code for the purposes of training.

The update pistols.dm commit doesn't actually do anything, thankfully. It's overwritten by the proper Covenant Training Pistol commit.

Please God let this be the last time.

Commit:
Update pistols.dm

Commit:
Covenant Training Pistol

Please God let this be the last time, it compiles.

Commit:
Update pistols.dm

Commit:
Revert "Update pistols.dm"

This reverts commit 5e04f1116b54808b34ffdf284c3be78be21fe7d1.

X0-11 commented on code/modules/halo/weapons/covenant/pistols.dm:
```diff
@@ -58,8 +58,12 @@
 	desc = "A dual funtionality pistol: It fires bolts of plasma, and when overcharged is capable of emitting a small emp burst at the point of impact. This one appears to be disabled"
 	max_shots = 0
 
-/obj/item/weapon/gun/energy/plasmapistol/disabled/attack_self(var/mob/user)
```
This shouldn't be removed.

Boltersam replied:
handled with the most recent updated one

And also, sorry that I haven't been naming my commits, it slipped my mind up to now

Commit:
Update pistols.dm

caelaislinn commented on code/modules/halo/weapons/covenant/pistols.dm:
```diff
@@ -59,7 +59,13 @@
 	max_shots = 0
 
 /obj/item/weapon/gun/energy/plasmapistol/disabled/attack_self(var/mob/user)
-	return
```
Although removing this line doesn't change anything, it's better not to make unnecessary changes.

caelaislinn commented on code/modules/halo/weapons/covenant/pistols.dm:
```diff
@@ -59,7 +59,13 @@
-	return
+
+/obj/item/weapon/gun/energy/plasmapistol/trainingpistol
+	name = "Type-25B Directed Energy Pistol"
```
It might be a good idea to give this a name like "Type-25B Directed Energy (Training) Pistol" so it doesn't share the same name as the actual plasma pistol

Boltersam replied:
Done and done.

caelaislinn reviewed (approved):
I have a few minor code suggestions but this is ready to merge regardless

Commit:
Adds Line, Renames Pistol

Title.
Readds the 'return' line that was removed
Changes the name from "Type-25B Directed Energy Pistol" to "Type-25B Directed Energy (Training) Pistol"

caelaislinn commented:
Closing and reopening to trigger a new travis run

caelaislinn commented:
This can be merged, the Travis failure is due to a Geminus issue introduced in #1302 and I have a fix waiting in #1303
